.github/workflows/build.yml

```diff
@@ -0,0 +1,145 @@
+name: build
+
+# We now default to running this workflow on every push to every branch.
+# This provides fast feedback when build issues occur, so they can be
+# fixed prior to being merged to the main branch.
+#
+# If you want to opt out of this, and only run the build on certain branches
+# please refer to the documentation on branch filtering here:
+#
+#   https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore
+#
+#on: [workflow_dispatch, push]
+
+# By dispatch only in development
+on: [workflow_dispatch]
+
+env:
+  PKG_NAME: "terraform-provider-cloudinit"
+
+jobs:
+  get-go-version:
+    name: "Determine Go toolchain version"
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.go-version }}
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - name: Determine Go version
+        id: get-go-version
+        # We use .go-version as our source of truth for current Go
+        # version, because "goenv" can react to it automatically.
+        # Specify the exact Go version (e.g., 1.22.4) in .go-version
+        # if your build requires a specific patch version of Go.
+        # Otherwise, specify the major and minor versions (e.g., 1.22),
+        # with a caveat that it can lead to builds using different
+        # patch versions of Go in a workflow run.
```

Comment:
Interesting! I wonder if now that the Go toolchain defaults to including patch version details in the `go.mod`, if we really need this static file in the future 🤔